nomos-da/network/messages/src/packing.rs

```diff
@@ -0,0 +1,156 @@
+// STD
+use futures::AsyncReadExt;
+use std::io;
+// Crates
+use nomos_core::wire;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+// Internal
+use crate::Result;
+
+type LenType = u16;
+const MAX_MSG_LEN_BYTES: usize = size_of::<LenType>();
+const MAX_MSG_LEN: usize = 1 << (MAX_MSG_LEN_BYTES * 8);
+
+fn into_failed_to_serialize(error: wire::Error) -> io::Error {
+    io::Error::new(
+        io::ErrorKind::InvalidData,
+        format!("Failed to serialize message: {}", error),
+    )
+}
+
+fn into_failed_to_deserialize(error: wire::Error) -> io::Error {
+    io::Error::new(
+        io::ErrorKind::InvalidData,
+        format!("Failed to deserialize message: {}", error),
+    )
+}
+
+fn get_message_too_large_error(message_len: usize) -> io::Error {
+    io::Error::new(
+        io::ErrorKind::InvalidData,
+        format!(
+            "Message too large. Maximum size is {}. Actual size is {}",
+            MAX_MSG_LEN, message_len
+        ),
+    )
+}
+
+pub fn pack<Message>(message: &Message) -> Result<Vec<u8>>
```

Comment:
Uhm, I'm not sure if should not go back to the trailing `_message` in this and below methods naming.

Reply:
What's the benefit?

Reply:
it is more explicit on where to use it (?). As Message type is really anything that can be serializable, although the type name is a hint 🤔 . Ok, we can leave it for now I guess.